cardano-cli/src/Cardano/CLI/EraBased/Run/Transaction.hs

```diff
@@ -461,7 +458,7 @@ runTxBuildRaw era
       cardanoEraConstraints era $ createAndValidateTransactionBody era txBodyContent
 
 runTxBuild :: ()
-  => CardanoEra era
+  => ShelleyBasedEra era
```

Comment:
Changing this type eon to `ShelleyBasedEra` so that the `Byron` case which could never have worked doesn't need to be handled.